### ISSUE TITLE
Hugo parses all content files, regardless of ignoreFiles status, and fails build if parsing fails

### ISSUE THREAD

**Issue by seanhoughton** [gohugoio/hugo#10303]
### What version of Hugo are you using (`hugo version`)?

`hugo v0.102.3+extended darwin/arm64 BuildDate=unknown` installed with Homebrew

### Does this issue reproduce with the latest release?

Yes, I tested this with a clone+build from main branch.

### What

Hugo parses all files in the content tree, even if they match the `ignoreFiles` setting in config.toml. If you have a file in the content tree that matches an `ignoreFiles` regex and also contains something that the hugo parser believes is a syntax error your build will fail.

In my specific case, I have a python virtualenv folder next to one of my markdown content files which I used to manually generate some of the images that are referenced. Some python site-packages contain html template files which start with the `{` character and Hugo incorrectly interprets this as a JSON file due to the first character in the file and then fails to unmarshal it. I have added the virtualenv to the ignoreFiles section but this happens before the ignore list is even considered in the code.

This seems to be related to the `mapContentForResult(...)` function in page.go

#### Repro

Configure hugo to ignore "ignoreme" folders:

```
ignoreFiles = ["ignoreme"]
```

Create an html file stating with a `{` 

```
 % cat content/blog/sample/ignoreme/badfile.html 
{
bug
```

Place it in the content tree like so
```
├── content
│   └── blog
│       ├── index.md
│       └── sample
│           ├── ignoreme
│           │   └── badfile.html
│           └── index.md
```

Attempt to serve
```
% hugo serve
Start building sites … 
hugo v0.102.3+extended darwin/arm64 BuildDate=unknown
Error: Error building site: "/tmp/bug/content/blog/sample/ignoreme/badfile.html:1:1": unexpected EOF parsing JSON front matter
```

**Comment by abaruchi** [comment]
Looks like in the newer version this is fixed. Just tested with the following version and worked fine.

```
$ hugo version
hugo v0.105.0-DEV-26eb93a54e336964bcda3a94a171518d8c79b45b darwin/amd64 BuildDate=2022-10-13T21:41:44Z
``
`

**Comment by stackkrocket** [comment]
This seems to be fixed with the current version

**Comment by jmooring** [comment]
This is not fixed in v0.104.3. Minimal reproducible example:

```bash
git clone --single-branch -b hugo-github-issue-10303 https://github.com/jmooring/hugo-testing hugo-github-issue-10303
cd hugo-github-issue-10303
hugo
```

**Comment by jmooring** [comment]
Workaround:

```
[[module.mounts]]
source = 'content'
target = 'content'
excludeFiles = '**ignoreme'
```
